tools/src/main/java/org/apache/kafka/tools/ProducerPerformance.java

```diff
@@ -127,12 +122,18 @@ public static void main(String[] args) throws Exception {
             int currentTransactionSize = 0;
             long transactionStartTime = 0;
             for (long i = 0; i < numRecords; i++) {
+
+                if (recordSize != null) {
+                    payload = new byte[recordSize];
+                    for (int j = 0; j < payload.length; ++j)
+                        payload[j] = (byte) (random.nextInt(26) + 65);
+                }
+
                 if (transactionsEnabled && currentTransactionSize == 0) {
                     producer.beginTransaction();
                     transactionStartTime = System.currentTimeMillis();
                 }
 
-
                 if (payloadFilePath != null) {
                     payload = payloadByteList.get(random.nextInt(payloadByteList.size()));
                 }
```

Comment:
I think the random payload generation should be the `else` case here. Like this:

```java
if (payloadFilePath != null) {
    payload = payloadByteList.get(random.nextInt(payloadByteList.size()));
} else if (recordSize != null) {
    payload = new byte[recordSize];
    ....
} else {
    throw new IllegalArgumentException("no payload File Path or record Size provided");
}
```

Reply:
👍